client/src/components/TodoFilter.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from "react";
+import {
+  IconButton,
+  Menu,
+  MenuItem,
+  FormControl,
+  Select,
+  Checkbox,
+  ListItemText,
+  OutlinedInput,
+  SelectChangeEvent,
+} from "@mui/material";
+import FilterListIcon from "@mui/icons-material/FilterList";
+import { useSearchParams } from "react-router-dom";
+import { TODO_TYPES } from "../constants/constants";
+
+interface TodoFiltersProps {
+  setTypes: (types: string[]) => void;
+}
+
+const TodoFilterMenu: React.FC<TodoFiltersProps> = ({ setTypes }) => {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const typeParam = searchParams.get("type");
+
+  const initialTypes =
+    typeParam === "[]"
+      ? TODO_TYPES
+      : typeParam
+      ? (() => {
```

Comment:
can we somehow create that function outside the variable (probably wrap the function into the useCallback) and call it inside the variable